tests/python/unittest/test_while_loop.py

```diff
@@ -0,0 +1,969 @@
+            max_iterations=10,
+        )
+        if hybridize:
+            model.hybridize(inline_limit=0)
+        _, result = model(
+            mx.nd.array([1], dtype="int64"), # i
+            mx.nd.array([0], dtype="int64"), # s
+        )
+        assert result[0].asscalar() == 6
+        assert result[1].asscalar() == 15
+        # Case 1.2: result should be sum([1, 2, 3 ... 1000])
+        model = _TestBlock(
+            cond=lambda i, s, true: true,
+            func=lambda i, s, true: (None, (i + 1, s + i, true)),
+            max_iterations=1000,
+        )
+        if hybridize:
+            model.hybridize(inline_limit=0)
+        _, result = model(
+            mx.nd.array([1], dtype="int64"), # i
+            mx.nd.array([0], dtype="int64"), # s
+            mx.nd.array([1], dtype="int64"), # true
+        )
+        assert result[0].asscalar() == 1001
+        assert result[1].asscalar() == 500500
+        assert result[2].asscalar() == 1
+        # Case 1.3: result should be sum([])
+        model = _TestBlock(
+            cond=lambda i, s, false: false,
+            func=lambda i, s, false: (None, (i + 1, s + i, false)),
+            max_iterations=1000,
+        )
+        if hybridize:
+            model.hybridize(inline_limit=0)
+        _, result = model(
+            mx.nd.array([1], dtype="int64"), # i
+            mx.nd.array([0], dtype="int64"), # s
+            mx.nd.array([0], dtype="int64"), # false
+        )
+        assert result[0].asscalar() == 1
+        assert result[1].asscalar() == 0
+        assert result[2].asscalar() == 0
+        # Case 2.1: result should be sum([1, 2, 3 ... 100])
+        model = _TestBlock(
+            cond=lambda i, s: i <= 100,
+            func=lambda i, s: (i, (i + 1, s + i)),
+            max_iterations=1000,
+        )
+        if hybridize:
+            model.hybridize(inline_limit=0)
+        (outputs, ), (result_i, result_s) = model(
+            mx.nd.array([1], dtype="int64"), # i
+            mx.nd.array([0], dtype="int64"), # s
+        )
+        assert all(outputs.asnumpy()[ : 100] == np.arange(1, 101).reshape(100, 1))
+        assert result_i.asscalar() == 101
+        assert result_s.asscalar() == 5050
+        # Case 2.2: result should be sum([1, 2, 3 ... 1000])
+        model = _TestBlock(
+            cond=lambda i, s, true: true,
+            func=lambda i, s, true: (i, (i + 1, s + i, true)),
+            max_iterations=1000,
+        )
+        if hybridize:
+            model.hybridize(inline_limit=0)
+        (outputs, ), (result_i, result_s, _) = model(
+            mx.nd.array([1], dtype="int64"), # i
+            mx.nd.array([0], dtype="int64"), # s
+            mx.nd.array([1], dtype="int64"), # true
+        )
+        assert all(outputs.asnumpy() == np.arange(1, 1001).reshape(1000, 1))
+        assert result_i.asscalar() == 1001
+        assert result_s.asscalar() == 500500
+        # Case 2.3: very corner case
```

Comment:
what corner case does it test?

Reply:
The loop body is never executed. Thanks for the review, I change the comment to "Case 2.3: a corner case, in which loop body is never executed" to make it clear.